client/state/ui/editor/video-editor/reducer.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	VIDEO_EDITOR_RESET_STATE,
+	VIDEO_EDITOR_SET_POSTER_URL,
+	VIDEO_EDITOR_SHOW_ERROR,
+	VIDEO_EDITOR_SHOW_UPLOAD_PROGRESS,
+} from 'state/action-types';
+
+/**
+ * Tracks poster updated state.
+ *
+ * @param {Boolean} state Whether or not the video poster was already updated
+ * @param {Object} action Action object
+ * @returns {Boolean} Whether or not the video poster is now updated
+ */
+export const isPosterUpdated = ( state = false, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_SET_POSTER_URL:
+		case VIDEO_EDITOR_SHOW_ERROR:
+			return true;
+
+		case VIDEO_EDITOR_RESET_STATE:
+			return false;
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster URL state.
+ *
+ * @param  {String} state Current poster URL
+ * @param  {Object} action Action object
+ * @return {String} Updated poster URL
+ */
+export const posterUrl = ( state = '', action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_SET_POSTER_URL:
+			return action.posterUrl;
+
+		case VIDEO_EDITOR_RESET_STATE:
+			return '';
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster upload progress state.
+ *
+ * @param  {Number} state Current upload progress of the poster
+ * @param  {Object} action Action object
+ * @return {Number} Updated upload progress of the poster
+ */
+export const uploadProgress = ( state = 0, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_SHOW_UPLOAD_PROGRESS:
+			return action.percentage;
```

Comment:
some advice I'd share from the importer is that in some cases 100% is a value we never really see and it might be beneficial to fib a bit here. since 100% came at the very end, or never came because the `load` event occurred after something close like 96% I made it look like 100% names at an early value and this meant that people would see a positive confirmation that the processes was ending vs going from _processing_ to _oops I missed the end_.

this can also (maybe more appropriately) be handled in the UI